app/src/main/AndroidManifest.xml

```diff
@@ -33,6 +33,12 @@
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.SEND" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="image/*" />
+            </intent-filter>
```

Comment:
You attached this on MainActivity, not ScanActivity

Reply:
fixed